typescript/provider-sdk/src/module.ts

```diff
@@ -30,6 +30,20 @@ export interface HypModule<M extends ModuleType> {
   update(config: Config<M>): Promise<AnnotatedTx[]>;
 }
 
+export interface ArtifactType {
+  config: unknown;
+  addresses: AddrMap;
+}
+
+export interface ArtifactReader<A extends ArtifactType> {
+  read(address: string): Promise<A['config']>;
```

Comment:
shouldn't it be `read(address: string): Promise<WithAddress<A['config']>>;` here?

Reply:
mmm i'm thinking about the composite artifacts as it might be useful for the nested artifacts to have their address in the config like we do right now, i don't see how the current API might help in case we need that as reading would not populate the address field anymore 🤔